src/main/kotlin/lotto/domain/ProfitRate.kt

```diff
@@ -0,0 +1,31 @@
+package lotto.domain
+
+import java.math.RoundingMode
+import java.text.DecimalFormat
+
+class ProfitRate(
+    private val matchResult: Map<Rank, Int>,
+    private val inputPrice: Int,
+) {
+    fun calculate(): Double {
+        val totalWinningMoney = getTotalWinningMoney()
+        return calculate(totalWinningMoney)
+    }
+
+    private fun getTotalWinningMoney(): Int {
+        return matchResult.keys
+            .sumOf { (matchResult[it] ?: DEFAULT_ZERO) * it.winningMoney }
```

Comment:
이 부분은 아래처럼 map에서 직접 get([])을 하는 부분을 없앨 수 있습니다.
```suggestion
        return matchResult.entries
            .sumOf { it.key.winningMoney * it.value }
```
Rank의 상금에 직접 접근하는 것 보다는, Rank에 메시지를 던져서 총액을 반환하는 형태도 좋아 보입니다 🙂

Reply:
좋네요ㅎㅎ
피드백 주신 내용이 더 깔끔한 코드인 것 같습니다.
추가로 Rank에 메세지를 던지는 방향으로 개선했습니다.